spring-integration-core/src/test/java/org/springframework/integration/codec/kryo/CompositeCodecTests.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2013 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.codec.kryo;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.springframework.integration.codec.Codec;
+import org.springframework.integration.codec.CompositeCodec;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author David Turanski
+ */
+public class CompositeCodecTests {
+
+	private Codec codec;
+
+
+	@SuppressWarnings({"unchecked", "rawtypes"})
+	@Before
+	public void setup() {
+		Map<Class<?>, Codec> codecs = new HashMap<Class<?>, Codec>();
+		codec = new CompositeCodec(codecs, new PojoCodec());
+	}
+
+	@Test
+	public void testPojoSerialization() throws IOException {
+		SomeClassWithNoDefaultConstructors foo = new SomeClassWithNoDefaultConstructors("hello", 123);
+		SomeClassWithNoDefaultConstructors foo2 = (SomeClassWithNoDefaultConstructors) codec.deserialize(
+				codec.serialize(foo),
+				SomeClassWithNoDefaultConstructors.class);
```

Comment:
Well since we're really only testing PojoCodec it returns an object. We could use generic here for the method similar to context.getBean(Foo.class).  I'll take a look